geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/netty/Coder.java

```diff
@@ -332,9 +343,55 @@ public static BigDecimal bytesToBigDecimal(byte[] bytes) {
   }
 
   public static long bytesToLong(byte[] bytes) {
-    return Long.parseLong(bytesToString(bytes));
+    return parseLong(bytes);
+  }
+
+  private static NumberFormatException createNumberFormatException(byte[] bytes) {
+    return new NumberFormatException("For input string: \"" + bytesToString(bytes) + "\"");
+  }
+
+  /**
+   * This method was derived from openjdk Long.java parseLong
+   */
+  public static long parseLong(byte[] bytes) throws NumberFormatException {
+    final int len = bytes.length;
+    if (len <= 0) {
+      throw createNumberFormatException(bytes);
+    }
+    int i = 0;
+    long limit = -Long.MAX_VALUE;
+    boolean negative = false;
+    byte firstByte = bytes[0];
+    if (firstByte < NUMBER_0_BYTE) { // Possible leading "+" or "-"
+      if (firstByte == bMINUS) {
+        negative = true;
+        limit = Long.MIN_VALUE;
+      } else if (firstByte != bPLUS) {
```

Comment:
If we want to match Redis' behaviour exactly, we should also throw an exception if we're given a number that starts with "+" or with "-0". These could be done as separate tickets though, if you feel it's not worth including them here. In fact, we already have [GEODE-9482](https://issues.apache.org/jira/browse/GEODE-9482) to track the second case.

Reply:
lets do that on another ticket. It does seem we might get some other benefits of having our own code to do the conversion instead of just using the jdk.